src/main/java/org/eclipse/golo/cli/command/GoloGoloCommand.java

```diff
@@ -78,7 +78,7 @@ public void execute() throws Throwable {
       }
     } else if (file.getName().endsWith(".golo")) {
       try (FileInputStream in = new FileInputStream(file)) {
-        Class<?> loadedClass = loader.load(file.getName(), in);
+        Class<?> loadedClass = loader.load(file.getName(), in, file.toURI().toURL());
```

Comment:
Build the `CodeSource` instance here (see the proposed `load` refactoring). This will allows to construct a `Certificate[]` from command line options if we ever need to.
BTW, the file scanning facility should be extracted.

Reply:
@yloiseau what do you mean by "extracted"?

Reply:
There should be a mode generic method that scan (recursively) a directory and return golo files, maybe in the form of a `CodeSource` object, or `Path` with a method to convert a `Path` into a `CodeSource`

Reply:
see the "extract method" refactoring...